components/brave_rewards/core/engine/hash_prefix_store.cc

```diff
@@ -128,45 +114,57 @@ bool HashPrefixStore::UpdatePrefixes(const std::string& prefixes,
 
   Close();
 
-  if (!IsValidPrefixSize(prefix_size)) {
-    return false;
-  }
-
   auto prefix_count = GetPrefixCount(prefixes.length(), prefix_size);
   if (!prefix_count) {
     return false;
   }
 
-  base::File file;
-  file.Initialize(file_path_,
-                  base::File::FLAG_CREATE_ALWAYS | base::File::FLAG_WRITE);
-  if (!file.IsValid()) {
-    return false;
-  }
+  flatbuffers::FlatBufferBuilder builder;
 
-  auto encoded_version = base::U32ToLittleEndian(kFileVersion);
-  if (!file.WriteAtCurrentPosAndCheck(base::as_byte_span(encoded_version))) {
-    return false;
-  }
+  std::vector<std::vector<char>> suffix_arrays(kOffsetsSize);
+  std::array<uint32_t, kOffsetsSize> offsets = {};
 
-  auto encoded_size =
-      base::U32ToLittleEndian(base::checked_cast<uint32_t>(prefix_size));
-  if (!file.WriteAtCurrentPosAndCheck(base::as_byte_span(encoded_size))) {
-    return false;
+  const std::string_view prefixes_view(prefixes);
+
+  for (size_t i = 0; i < prefix_count.value(); i++) {
+    const auto prefix = prefixes_view.substr(i * prefix_size, prefix_size);
```

Comment:
This is pretty tricky code. TBH, I'm leaning towards preferring the simpler current approach for now. What would you think about leaving as-is for a couple of releases, and then introducing the optimization later, if necessary?

Also, since we're changing the file format, we're going to need another pref migration to reset the fetch timer.

Reply:
Okay, let's postpone this for some time.
IMHO, the most tricky part here is flattering (because we don't want to use variable sites flatbuffers vector).
If we stick to a custom format, this could simplify code a little bit (and also remove reinterpret_casts)